package/health_daemon/src/gnss_bias_monitor.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <libpiksi/logging.h>
+#include <libpiksi/sbp_zmq_pubsub.h>
+
+#include <libsbp/sbp.h>
+#include <libsbp/observation.h>
+
+#include "health_monitor.h"
+#include "utils.h"
+
+#include "gnss_bias_monitor.h"
+
+#define GNSS_BIAS_ALERT_RATE_LIMIT (240000) /* ms */
+
+static health_monitor_t *gnss_bias_monitor;
+
+static struct gnss_bias_ctx_s {
+  bool gnss_read_resp;
+  bool gnss_enabled;
+} gnss_bias_ctx = {
+  .gnss_read_resp = false,
+  .gnss_enabled = false
+};
+
+static void sbp_msg_read_resp_callback(u16 sender_id,
+                                       u8 len, u8 msg_[], void *ctx)
+{
+  (void)sender_id;
+  (void)len;
+  health_monitor_t *monitor = (health_monitor_t *)ctx;
+
+  const char *section, *name, *value;
+  if(health_util_parse_setting_read_resp(msg_, len,
+                                         &section, &name, &value) == 0) {
+    bool last_gnss_enabled = gnss_bias_ctx.gnss_enabled;
+    if (health_util_check_gnss_enabled(section, name, value,
+                                       &gnss_bias_ctx.gnss_enabled) == 0) {
+      gnss_bias_ctx.gnss_read_resp = true;
+      if (gnss_bias_ctx.gnss_enabled && !last_gnss_enabled) {
+        health_monitor_reset_timer(monitor);
+      }
+    }
+  }
+}
+
+static int gnss_bias_timer_callback(health_monitor_t *monitor, void *context)
+{
```

Comment:
Is this check specific to GLONASS?  Or does it apply to GPS too?

Reply:
Based on my conversations with @anth-cole the check should be active only when the user-based setting for glonass is enabled. I inferred that setting to be acquisition.glonass_acquisition_enabled